api/pkg/apis/v1alpha1/vendors/catalogs-vendor.go

```diff
@@ -295,6 +295,15 @@ func (e *CatalogsVendor) onCatalogs(request v1alpha2.COARequest) v1alpha2.COARes
 
 	lLog.Infof("V (Catalogs Vendor): onCatalogs, method: %s, traceId: %s", string(request.Method), span.SpanContext().TraceID().String())
 
+	id, err := api_utils.GetUnescapaedString(request.Parameters["__name"])
```

Comment:
Personally, I would prefer you can write a common function to every vendor, and pass the logger, vendor name, function and sth like that. One example is on one day, if we want to change the status code and return status code, we need to change every vendor.

Reply:
This seems common function not that common.